internal/xds/translator/teleport_translate_test.go

```diff
@@ -0,0 +1,54 @@
+package translator
+
+import (
+	"testing"
+
+	infra "github.com/envoyproxy/gateway/internal/infrastructure/kubernetes"
+	resourcev3 "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTeleportTranslateXds(t *testing.T) {
```

Comment:
Marking things with "Teleport" is useful when we don't plan to upstream them. Wouldn't we want to upstream this functionality/test?

If this is functionality that we never plan to upstream, how do plan to move off this fork in the future? Should we make this configurable instead?